File: Harvard-CS50x/pset6/vigenere.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("missing command-line argument")
        exit(0)
    
    key = sys.argv[1]
    keylen = len(key)
    if sys.argv[1].isalpha() == False:
        print("Only letters allowed!")
        exit(1)
    
    print("plaintext: ", end="")
    plaintext = cs50.get_string()

    ciphertext = []
    j = 0
    for i in range(len(plaintext)):
        if j >= keylen:
            j = 0
        k = ord(key[j]) % 97
        c = ord(plaintext[i])
        if plaintext[i].isalpha()==True:
            if plaintext[i].isupper()==True:
                c = (c + k - 65) % 26 + 65
                ciphertext.append(chr(c))
                j += 1
            if plaintext[i].islower()==True:
                c = (c + k - 97) % 26 + 97
                ciphertext.append(chr(c))
                j += 1            
        else:
            ciphertext.append(chr(c))
    
    print("ciphertext: ", end="")
    print("".join(ciphertext))
    exit(2)
    
if __name__ == "__main__":
    main()

File: Harvard-CS50x/pset2/vigenere.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/**
 * vigenere.c
 *
 * Encrypts messages using the Vigenere cipher. Retrieves the message to encrypt via user prompt.
 * 
 * Usage: ./vigenere keyword
 * 
 * where keyword is the cipher keyword that is entirely composed of alphabetical characters.
 * 
 * salimt
 */

int main(int argc, string argv[]) {
    if (argc != 2) {
        printf("missing key argument\n");
        return 1;
    }
    string key = argv[1];
    int klen = strlen(key);
    printf("%lu is the length of the key\n", strlen(key));
    for (int i = 0; i < klen; i++) {
        if (!isalpha((unsigned char)key[i])) {
            printf("only letters allowed\n");
            return 1;
        }
        key[i] = tolower((unsigned char)key[i]) - 'a';
    }
    printf("Please enter the phrase you would like to encrypt: ");
    string txt = GetString();
    printf("%lu is the length of the phrase.\n", strlen(txt));
    printf("Encrypted: ");
    for (int i = 0, j = 0; i<strlen(txt); i++) {
        int c = (unsigned char)txt[i];
        if (isupper(c)) {
            c = (c - 'A' + key[j++ % klen]) % 26 + 'A';
        }if (islower(c)) {
            c = (c - 'a' + key[j++ % klen]) % 26 + 'a';
        }putchar(c);
    }
    putchar('\n');
    return 0;
}